src/main/java/com/znsio/rpap/utils/DriverFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.znsio.rpap.utils;
+
+import org.apache.log4j.Logger;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+public class DriverFactory {
+    private static final Logger LOGGER = Logger.getLogger(DriverFactory.class.getName());
+
+    public static WebDriver launchWebApplication(WebDriver driver, String browserName, String appURL) {
+
+        if (browserName.equalsIgnoreCase("chrome")) {
```

Comment:
With Selenium 4, you do not need to provide drivers. Selenium will download them automatically based on the browser version available on the test execution machine

Reply:
Updated